src/PagesController.php

```diff
@@ -0,0 +1,276 @@
+<?php
+/**
+ * Pages Controller
+ *
+ * @author    Pronamic <info@pronamic.eu>
+ * @copyright 2005-2023 Pronamic
+ * @license   GPL-3.0-or-later
+ * @package   Pronamic\WordPress\Pay
+ */
+
+namespace Pronamic\WordPress\Pay;
+
+/**
+ * Pages Controller class
+ */
+class PagesController {
+	/**
+	 * Setup.
+	 */
+	public function setup() {
+		\add_action( 'init', [ $this, 'init' ] );
+
+		\add_action( 'admin_init', [ $this, 'admin_init' ] );
+	}
+
+	/**
+	 * Initialize.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		// Pages.
+		$pages = $this->get_pages();
+
+		foreach ( $pages as $id => $label ) {
+			\register_setting(
+				'pronamic_pay',
+				$id,
```

Comment:
Settings are not saved, because `$id` is an integer instead of the option name.

```suggestion
		foreach ( $pages as $page ) {
			\register_setting(
				'pronamic_pay',
				$page['option_name'],
```